src/main/scala/cromwell/engine/workflow/ExecutionStoreKey.scala

```diff
@@ -0,0 +1,66 @@
+package cromwell.engine.workflow
+
+import cromwell.binding._
+import cromwell.engine.ExecutionStatus
+import cromwell.engine.workflow.WorkflowActor.ExecutionStore
+
+sealed trait ExecutionStoreKey {
+  val scope: Scope
+
+  val index: Option[Int]
+
+  val parent: Option[ExecutionStoreKey]
+}
+
+
+case class CallKey(override val scope: Call,
+                   override val index: Option[Int],
+                   override val parent: Option[ExecutionStoreKey])
+  extends ExecutionStoreKey
+
+
+case class ScatterKey(override val scope: Scatter,
+                      override val index: Option[Int],
+                      //override val status: ExecutionStatus,
+                      override val parent: Option[ExecutionStoreKey])
+  extends ExecutionStoreKey {
+
+  /**
+   * Creates a sub-ExecutionStore with Starting entries for each of the scoped children.
+   * @param count Number of ways to scatter the children.
+   * @return ExecutionStore of scattered children.
+   */
+  def populate(count: Int): ExecutionStore = {
+    val keys = for {
+      childScope <- this.scope.children
+      collector = collectorForChild(childScope, count)
+      entry <- collector +: collector.keys
+    } yield entry
+
+    (keys map {
+      _ -> ExecutionStatus.Starting
+    }).toMap
```

Comment:
I'm not sure if setting the status to `Starting` is correct.  Shouldn't it be `NotStarted`?